pkg/bundles/kontext/bundle.go

```diff
@@ -44,60 +47,95 @@ func bundle(directory string) (v1.Layer, error) {
 	tw := tar.NewWriter(buf)
 	defer tw.Close()
 
+	excludedDirs := sets.NewString()
+	includesDirs := sets.NewString()
+
 	err := filepath.Walk(directory,
 		func(path string, fi os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-
-			// Skip anything in the .git directory
-			// TODO(mattmoor): expand this to .gitignore / .dockerignore?
-			if fi.IsDir() && filepath.Base(path) == ".git" {
-				return filepath.SkipDir
-			}
-
-			// Chase symlinks.
-			info, err := os.Stat(path)
-			if err != nil {
-				return err
-			}
-
-			// Compute the path relative to the base path
-			relativePath, err := filepath.Rel(directory, path)
-			if err != nil {
-				return err
-			}
-
-			newPath := filepath.Join(StoragePath, relativePath)
-
-			if info.Mode().IsDir() {
-				return tw.WriteHeader(&tar.Header{
+			//Always include the root directory
+			includesDirs.Insert(directory)
+			if directory != path {
+				if err != nil {
+					return err
+				}
+
+				//if parent is included then dont check them children, include them implicitly
+				var isParentIncluded bool
```

Comment:
Parents will be walked before children and we can avoid this complexity if we return the special error code: `filepath.SkipDir`

Reply:
Yes that's correct.  That points to a need to return more than a bool, so perhaps we should introduce an ~enum to signify how much to skip.